### PR TITLE
[dlinksmarthome] Reboot device daily following shutdown of cloud service

### PR DESCRIPTION
**Pull request by MikeJMajor**
Following the shutdown of the mydlink Home service, the devices have a tendency to become unresponsive after a few days.

This change performs a daily reboot of the device to keep it responsive.

Discovery has also been changed to avoid using ipv6 addresses which don't seem to work. The change is based on  a change made to the Chromecast binding. The binding previously had a problem in that it would occasionally switch to using an ipv6 address and go offline, rebooting daily increases the chance it ends up with an ipv6 address.

### CODE REVIEW

File: bundles/org.openhab.binding.dlinksmarthome/README.md [hunk 1]
```diff
@@ -6,11 +6,11 @@ A binding for D-Link Smart Home devices.
 
 ### DCH-S150 (WiFi motion sensor)
 
-The binding has been tested with hardware revisions A1 and A2 running firmware version 1.22.
+The binding has been tested with hardware revisions A1 and A2 running firmware version 1.22. The mydlink Home service is now end of life and the device requires a daily reboot (performed by the binding) to keep it responsive.
```
Comment (lolodomo) [inline bundles/org.openhab.binding.dlinksmarthome/README.md:9]: Minor: please go to a new line for each sentence.

File: bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/DLinkSmartHomeDiscoveryParticipant.java [hunk 2]
```diff
@@ -96,7 +97,11 @@ private ThingTypeUID getThingType(final ServiceInfo serviceInfo) {
 
     private DiscoveryResult createMotionSensor(final ThingUID thingUID, final ThingTypeUID thingType,
             final ServiceInfo serviceInfo) {
-        final String host = serviceInfo.getHostAddresses()[0];
+        final Inet4Address[] addresses = serviceInfo.getInet4Addresses();
+        if (addresses.length != 1) {
+            return null;
+        }
+        final String host = addresses[0].getHostAddress();
```
Comment (lolodomo) [inline bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/DLinkSmartHomeDiscoveryParticipant.java:104]: This is a change in the discovery that you even don't announce ?
I think this should be mentioned in the description of your PR.

File: bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkMotionSensorHandler.java [hunk 3]
```diff
@@ -64,6 +64,9 @@ public void sensorStatus(final DeviceStatus status) {
             case ONLINE:
                 updateStatus(ThingStatus.ONLINE);
                 break;
+            case REBOOTING:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DUTY_CYCLE, "Device rebooting");
```
Comment (lolodomo) [inline bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkMotionSensorHandler.java:68]: ThingStatusDetail.DUTY_CYCLE is I believe not so much used but could be the right choice there.
Only used in 3 bindings: homematic, lutron and MQTT generic. I will see how it is used bu others.

File: bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/motionsensor/DLinkMotionSensorConfig.java [hunk 4]
```diff
@@ -34,4 +33,9 @@ public class DLinkMotionSensorConfig {
      * The pin code of the device
      */
     public String pin;
+
+    /**
+     * The hour to reboot the device
+     */
+    public Integer rebootHour;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/motionsensor/DLinkMotionSensorConfig.java:40]: You can simply use `int` type.

**Review by lolodomo — CHANGES_REQUESTED** [review]
Partial review

File: bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/DLinkSmartHomeDiscoveryParticipant.java (hunk 2 above)
Comment (MikeJMajor) [reply bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/DLinkSmartHomeDiscoveryParticipant.java:104]: Description updated to include why this change was made.

File: bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkMotionSensorHandler.java (hunk 3 above)
Comment (MikeJMajor) [reply bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkMotionSensorHandler.java:68]: Are you happy with the use of DUTY_CYCLE?

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

File: bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkMotionSensorHandler.java (hunk 3 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkMotionSensorHandler.java:68]: Let's assume it is OK.

**Pull request merged**